cfspopcon/key_checker.py

```diff
@@ -0,0 +1,195 @@
+"""A simple tool to ensure consistency between the algorithms, default units and the physics glossary, with a CLI."""
+
+import re
+from importlib.resources import as_file, files
+from pathlib import Path
+
+import click
+import yaml
+
+from cfspopcon import Algorithm
+
+
+class KeyChecker:
+    """A class for comparing the keys in algorithms, default units and the physics glossary."""
+
+    def __init__(self) -> None:
+        """Read in the keys from the algorithms, default units and the physics glossary."""
+        self.glossary, self.glossary_keys = self.read_physics_glossary()
+        self.algorithm_keys = self.read_algorithm_keys()
+        self.default_unit_keys = self.read_default_unit_keys()
+
+        self.all_keys = self.algorithm_keys | self.glossary_keys | self.default_unit_keys
+
+    def run(self, dry_run: bool = True) -> bool:
+        """Check the files and, if dry_run = False, modify the files in place."""
+        issue_detected = False
+
+        issue_detected = self.fix_unit_dictionary(dry_run) or issue_detected
+        issue_detected = self.fix_glossary(dry_run) or issue_detected
+
+        return issue_detected
+
+    def read_physics_glossary(self) -> tuple[dict[str, list[str]], set[str]]:
+        """Read the physics glossary file."""
+        with as_file(files("cfspopcon").parents[0] / "docs" / "doc_sources" / "physics_glossary.rst") as filepath:  # type:ignore[attr-defined]
```

Comment:
If we move the script you can just assume that the script is run from the repo, and won't need to use `as_file` patterns.